From 2cc795c700d59c822e6d832d03e5dd70a1ec7656 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Thu, 19 Sep 2024 19:04:30 +0530
Subject: [PATCH] firmware: scm: Add support to check if fuse row read scm are
 available

Add an interface to check if fuse row read scm is supported or not.

Change-Id: Ib258d84e939204e308c439d695344d96b9dee816
Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
---
 drivers/firmware/qcom_scm.c            | 13 +++++++++++++
 include/linux/firmware/qcom/qcom_scm.h |  1 +
 2 files changed, 14 insertions(+)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index e3a64d289638..bebe332b50e5 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -1478,6 +1478,19 @@ bool qcom_qfrom_fuse_row_write_available(void)
 }
 EXPORT_SYMBOL_GPL(qcom_qfrom_fuse_row_write_available);
 
+/**
+ * qcom_qfrom_fuse_row_read_available() - is the fuse row read interface
+ *                                        available ?
+ *
+ * Return: true if the SCM call is supported
+ */
+bool qcom_qfrom_fuse_row_read_available(void)
+{
+	return __qcom_scm_is_call_available(__scm->dev, QCOM_SCM_SVC_FUSE,
+					    QCOM_QFPROM_ROW_READ_CMD);
+}
+EXPORT_SYMBOL_GPL(qcom_qfrom_fuse_row_read_available);
+
 /**
  * qcom_scm_ice_invalidate_key() - Invalidate an inline encryption key
  * @index: the keyslot to invalidate
diff --git a/include/linux/firmware/qcom/qcom_scm.h b/include/linux/firmware/qcom/qcom_scm.h
index e8e3c6b78847..6e422f331386 100644
--- a/include/linux/firmware/qcom/qcom_scm.h
+++ b/include/linux/firmware/qcom/qcom_scm.h
@@ -295,6 +295,7 @@ extern bool qcom_scm_ice_hwkey_available(void);
 extern bool qcom_qfprom_show_auth_available(void);
 extern bool qcom_sec_dat_fuse_available(void);
 extern bool qcom_qfrom_fuse_row_write_available(void);
+extern bool qcom_qfrom_fuse_row_read_available(void);
 extern int qcom_scm_ice_invalidate_key(u32 index);
 extern int qcom_config_sec_ice(void *buf, int size);
 extern int qcom_scm_ice_set_key(u32 index, const u8 *key, u32 key_size,
-- 
2.34.1

